authentication/auth.go

```diff
@@ -250,7 +219,29 @@ func generateClaimsV2(claimValues map[string][]string, launcherSchema surveys.La
 		surveyMetadata["receipting_keys"] = receiptingKeys
 	}
 
-	getSurveyMetadataFromClaims(claimValues, data, claims, surveyMetadata)
+	/*
+		Combine the schema metadata with mandatory/required schema metadata that's not defined
+		in the schema itself in order to ensure the correct metadata is loaded into surveyMetadata
+		and prevent duplicates being entered.
+	*/
+	defaults := GetDefaultValues()
+	mandatoryClaims := getMandatatoryClaims(schema.SurveyType, defaults)
+	schema.Metadata = append(mandatoryClaims, schema.Metadata...)
+	for _, mandatoryMetadata := range mandatoryClaims {
+		present := false
+		for _, schemaMetadata := range schema.Metadata {
+			if mandatoryMetadata.Name == schemaMetadata.Name {
+				present = true
+				break
+			}
+		}
+		schema.Metadata = append(schema.Metadata, mandatoryMetadata)
```

Comment:
Slightly confused about the logic here - as we are appending to the metadata if it is present or not?

Reply:
I was also slightly confused to my logic here 😆 I've updated it and hopefully it makes more sense.